pkg/operator/ceph/object/controller.go

```diff
@@ -142,9 +144,116 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		}
 	}
 
+	// Watch Secrets secrets annotated for the object store
+	err = c.Watch(source.Kind[client.Object](mgr.GetCache(),
+		&corev1.Secret{TypeMeta: metav1.TypeMeta{Kind: "Secret", APIVersion: corev1.SchemeGroupVersion.String()}},
+		handler.EnqueueRequestsFromMapFunc(mapSecretToCR(mgr.GetClient())),
```

Comment:
There is a typed variant of this func: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/handler#TypedEnqueueRequestsFromMapFunc